manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/AbstractWorker.java

```diff
@@ -141,4 +161,7 @@ protected boolean isTimeoutExceeded(Work work, ManagedResource managedResource)
     protected abstract boolean isProvisioningComplete(T managedResource);
 
     protected abstract boolean isDeprovisioningComplete(T managedResource);
+
+    protected abstract T recordError(Work work, Exception e);
```

Comment:
@manstis With you now having the `error_id` and `error_uuid` on the `ManagedResource`, I think we can provide a generic implementation of this method rather than having to provide the implementation in each sub-class:

```
    private T recordError(Work work, Exception e) {
        String managedResourceId = getId(work);
        T managedResource = load(managedResourceId);
        BridgeErrorInstance bridgeErrorInstance = bridgeErrorHelper.getBridgeErrorInstance(e);
        managedResource.setErrorId(bridgeErrorInstance.getId());
        managedResource.setErrorUUID(bridgeErrorInstance.getUuid());
        return persist(managedResource);
    }
```

Reply:
Good spot. Changed.